### PR TITLE
Lobby Menu Overhaul

### PR DESCRIPTION
**Pull request by T1mbits**
Overhauling lobby select screen to be simpler/more straightforward to navigate as well as improving appearance (hopefully :P)

TODO:
- [x] Add/fix scroll list for lobby cards
- [x] Add creation menu & ability to create lobbies
- [x] Redesign cards (?) & ability to join respective lobbies
- [ ] Filter menu & filtering + searching lobbies
- [ ] Properly align menu elements and fix proportions
- [x] Make the side buttons actually work

OPTIONAL:
- [x] Add card ordering (A-Z, Z-A, by ping, by fullest/emptiest lobby)

### CODE REVIEW

**Comment by T1mbits** [comment]
Added/fixed scrolling for lobby cards, will add new example vid for each significant task completed

https://github.com/user-attachments/assets/0a561e73-5c2b-4643-a4aa-38a452335647

**Comment by youbitchoc** [comment]
does the scrollwheel work? how does controller interact with this?

**Comment by T1mbits** [comment]
when your mouse hovers over you can scroll with your scroll wheel, controller users have to use the scroll buttons as far as I’m aware. I don’t have a controller to test or do anything myself so I can’t really be sure

**Comment by youbitchoc** [comment]
> when your mouse hovers over you can scroll with your scroll wheel

neat :D

> I don’t have a controller to test

if keyboard controls are ok then it should be fine

File: Menu/LobbySelectMenu.cs [hunk 1]
```diff
@@ -46,16 +47,46 @@ public LobbySelectMenu(ProcessManager manager) : base(manager, RainMeadow.Ext_Pr
...
+                new LobbyInfo(fakeEndpoint, "Them", "Meadow", 6, true, 28),
+                new LobbyInfo(fakeEndpoint, "Person", "Story", 2147483647, false, 2147483647),
+                new LobbyInfo(fakeEndpoint, "The One Who Waits", "ArenaCompetitive", 11, true, 12),
+                new LobbyInfo(fakeEndpoint, "Fisher Price Pebbles (lmao)", "Meadow", 21, false, 26),
+                new LobbyInfo(fakeEndpoint, "notchoc", "Story", -7, true, -16),
+                new LobbyInfo(fakeEndpoint, "Glistening Sanctuaries", "Story", 4, true, 15),
+                new LobbyInfo(fakeEndpoint, "Lighthearted Appraisal", "Story", 21, true, 16),
+            }.ToArray();
+            test.FilterLobbies();
+            test.CreateCards();
```
Comment (6fears7) [inline Menu/LobbySelectMenu.cs:82]: Remove this line.

Comment (T1mbits) [reply Menu/LobbySelectMenu.cs:82]: LIES THIS IS LIES DO NOT LISTEN TO HIM HES LYING

**Comment by T1mbits** [comment]
something about being able to see the creation menu

https://github.com/user-attachments/assets/4fc89d99-cfe7-4aa3-a8e4-c20d0197f4ab

**Comment by 6fears7** [comment]
The scroll and filtering will be crucial components to our release. Good work on this so far

**Comment by T1mbits** [comment]
ok I think I broke all of the commit history with this because I'm still not sure how I'm supposed to sync branches and stuff on forks or whatever but idk it looks fixed to me

https://github.com/user-attachments/assets/678c7e99-5455-407e-ac8c-13da62de935e

**Comment by 6fears7** [comment]
commits look good. Just remove the large number of test lobbies in LocalMatchmakingManager. I'll test functionality here in a few.

**Review by henpemaz — APPROVED** [review]
Looks good at a glance, we can cleanup the fake lobbies later.

Lets get this in our next release

**Comment by henpemaz** [comment]
turns out this one should have been closed in favor of the new draft one

**Pull request merged**